### PR TITLE
 refactored react router and add route specs

### PR DESCRIPTION
**Pull request by meltedspork**
This is much nicer set up with React router with tests added to ensure we are providing correct routes

### CODE REVIEW

**Review by kratsg — COMMENTED** [review]
I don't really understand the changes but where is the Router and Switch now?

**Comment by meltedspork** [comment]
@kratsg  - `Router` is really just an `BrowserRouter` renamed and is now in `index.js` as main wrapper. And `Switch` seems to be unneeded right now as link appears to be working currently. The router was sort of re-archiutured where store is actually stored inside router instead so it should be easier to access routes stuff for `store` if needed.

**Comment by kratsg** [comment]
> @kratsg - Router is really just an BrowserRouter renamed and is now in index.js as main wrapper. And Switch seems to be unneeded right now as link appears to be working currently. The router was sort of re-archiutured where store is actually stored inside router instead so it should be easier to access routes stuff for store if needed.

How do we know if we want the Store to wrap the Router, or the Router to wrap the Store?

It shouldn't matter either way for the Router, because we can wrap any component with the HOC `withRouter` to get access to the router information.

**Comment by meltedspork** [comment]
Actually, after more researching. Seems like we are doing this wrong actually - see https://github.com/ReactTraining/react-router/tree/master/packages/react-router-redux.

It has proper example that I will give it a shot.

**Review by kratsg — APPROVED** [review]

**Comment by meltedspork** [comment]
states can be accessed via routerReducer as props

screen shot 2018-02-20 at 4 05 55 pm

File: src/redux/modules/auth/middleware.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ const authMiddleware = ({ dispatch }) => next => (action) => {
         } else {
           window.history.replaceState({}, document.title, '.');
           dispatch(loginSuccess(authResult));
+          // we can use "push" to control the path we want to redirect
```
Comment (kratsg) [inline src/redux/modules/auth/middleware.js:19]: how are we planning on doing redirects?

**Comment by kratsg** [comment]
> states can be accessed via routerReducer as props

What's an example of this? How do you access the routerReducer? Will need to update my branch that has the example of using `withRouter` soon...

**Comment by meltedspork** [comment]
> What's an example of this? How do you access the routerReducer? Will need to update my branch that has the example of using withRouter soon...

it depends on where you want to access it. if in component then it will be similar how we did with `state.authReducer.isAuthenticated` in here https://github.com/deafchi/signsfive-web/blob/develop/src/containers/navigator/submit-a-sign.jsx#L29

in short something along the line would be as in code block where you would require in
```
const mapStateToProps = state => ({
  location: state. routerReducer.location,
});
```

if only want `pathname`: can do 
```
const mapStateToProps = state => ({
  pathname: state. routerReducer.location.pathname,
});
```

but if you want somewhere else like middleware, it can be just simple `state.routerReducer.location.pathname` I believe which you can access by destructing the `store` such as 
```
const authMiddleware = ({ dispatch, state }) => next => (action) => {
});
```

**Comment by kratsg** [comment]
And supposedly can `dispatch(changeRoute)` somehow?

File: src/redux/modules/auth/middleware.js (hunk 1 above)
Comment (meltedspork) [reply src/redux/modules/auth/middleware.js:19]: see below `dispatch(push('/foo'));` for example.

Comment (meltedspork) [reply src/redux/modules/auth/middleware.js:19]: But im not married to this method. if you have something better I am all for it :)

Comment (kratsg) [reply src/redux/modules/auth/middleware.js:19]: where are the docs for this stuff???

**Pull request merged**